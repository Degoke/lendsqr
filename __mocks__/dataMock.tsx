import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
