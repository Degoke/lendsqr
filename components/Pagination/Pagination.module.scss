.pagination {
  display: flex;
  justify-content: space-between;
  margin-top: 1rem;
  flex-wrap: wrap;
  row-gap: 1rem;

  select {
    background: rgba(33, 63, 125, 0.1);
    border-radius: 4px;
    padding: 0.2rem;
    margin: 0 0.5rem;
  }

  ul {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    row-gap: 1rem;
  }

  &__button {
    border: none;
    margin: 0 0.1rem;
    padding: 0 0.5rem;
    background: none;
  }

  &__nextButton {
    background: rgba(33, 63, 125, 0.1);
    border-radius: 4px;
    border: none;
    margin: 0 0.1rem;
    padding: 0 0.5rem;
  }
}

.active {
  font-weight: 600;
}
