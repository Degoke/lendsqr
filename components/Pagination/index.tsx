import React, { useEffect, useState } from 'react'
import styles from './Pagination.module.scss'

type Props = {
  currentPage: number
  totalUsers: number
  numberPerPage: number
  changeNumberPerPage: (e: React.ChangeEvent<HTMLSelectElement>) => void
  changeCurrentPage: (page: number) => void
}

export default function Pagination({
  totalUsers,
  currentPage,
  numberPerPage,
  changeNumberPerPage,
  changeCurrentPage,
}: Props) {
  const [numberOfPages, setNumberOfPages] = useState<number>(0)

  useEffect(() => {
    setNumberOfPages(Math.round(totalUsers / numberPerPage))
  }, [totalUsers, numberPerPage])

  const handlePreviousPage = () => {
    if (currentPage > 1) {
      changeCurrentPage(currentPage - 1)
    }
  }

  const handleNextPage = () => {
    if (currentPage < numberOfPages) {
      changeCurrentPage(currentPage + 1)
    }
  }

  return (
    <div className={styles.pagination}>
      <div>
        <span>Showing</span>
        <select value={numberPerPage} onChange={changeNumberPerPage}>
          <option value={15}>15</option>
          <option value={20}>20</option>
          <option value={25}>25</option>
          <option value={50}>50</option>
          <option value={100}>100</option>
        </select>
        <span>Out of {totalUsers}</span>
      </div>
      <ul>
        <li>
          <button
            className={styles.pagination__nextButton}
            onClick={handlePreviousPage}
          >
            {'<'}
          </button>
          {Array.from({ length: numberOfPages }, (_, i) => i + 1).map(
            (page) => (
              <button
                key={page}
                className={`${styles.pagination__button} ${
                  currentPage === page && styles.active
                }`}
                onClick={() => changeCurrentPage(page)}
              >
                {page}
              </button>
            )
          )}

          <button
            className={styles.pagination__nextButton}
            onClick={handleNextPage}
          >
            {'>'}
          </button>
        </li>
      </ul>
    </div>
  )
}
