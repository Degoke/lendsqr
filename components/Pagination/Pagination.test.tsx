import { render, screen } from '@testing-library/react'
import Pagination from './index'
import { mockCard, mockImage } from '../../__mocks__/dataMock'

describe('Pagination tests', () => {
  it('renders Pagination component unchanged', () => {
    const { container } = render(
      <Pagination
        currentPage={0}
        totalUsers={0}
        numberPerPage={0}
        changeCurrentPage={jest.isMockFunction}
        changeNumberPerPage={jest.isMockFunction}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
