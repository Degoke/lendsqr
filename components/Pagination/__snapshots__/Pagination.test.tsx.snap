// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination tests renders Pagination component unchanged 1`] = `
<div>
  <div
    class="pagination"
  >
    <div>
      <span>
        Showing
      </span>
      <select>
        <option
          value="15"
        >
          15
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="25"
        >
          25
        </option>
        <option
          value="50"
        >
          50
        </option>
        <option
          value="100"
        >
          100
        </option>
      </select>
      <span>
        Out of 
        0
      </span>
    </div>
    <ul>
      <li>
        <button
          class="pagination__nextButton"
        >
          &lt;
        </button>
        <button
          class="pagination__nextButton"
        >
          &gt;
        </button>
      </li>
    </ul>
  </div>
</div>
`;
